modal/image.py

```diff
@@ -553,6 +624,23 @@ def build_dockerfile(version: ImageBuilderVersion) -> DockerfileSpec:
             context_mount=mount,
         )
 
+    def add_local_python_packages(self, *packages: Union[str, Path], copy: bool = False) -> "_Image":
+        """Attaches local Python packages to the container running the image
```

Comment:
"Attach" was an oversight when I renamed it again, thanks for spotting!

Yeah, adding third party packages or packages that have real install recipes via this method isn't intended, but I can see how people would do that. The main intent is for it to be used for directories that are part of the user's current python projet, and the interface has been using the module name as a convenience for pointing to that directory via the current sys.path. 

Basically a short form for: `add_local_dir(lookup_package("my_package"), filter=modal.file_filters.PyFiles)` (assuming we had `lookup_package()` and filters... 

I have separately been thinking that it would be nice to just point to a "project directory" when setting up the user's image, which is very similar to this but using file paths, and maybe that's a better way to do this?
The problem with that is that specifying file system paths in a portable way is so un-ergonomic for even simple cases, e.g:

E.g.
```py
Image.debian_slim().add_local_dir(
    Path(__file__).parent,
    remote_path="/root/some_root_package_name",
    condition=python_mount_condition
)
```

Preferably I'd want it to work like above with default arguments of using:
```py
Image.debian_slim().add_project_dir()  # use directory of current file as project dir - "standard", only include python files (or .modalignore?)
```

And then have some very easy ways to customize this:

```py
Image.debian_slim().add_project_dir(l"..")  # use parent directory as root, using relative paths from the current file's dir
```

```py
Image.debian_slim().add_project_dir(filter=modal.filters.AllFiles)  # use all files instead of just python files
```

Reply:
I'll rename `add_local_python_packages` to a "private `_add_local_python_packages` for now to get the rest of the changes merged in while we discuss the best way of "adding my local python *files*"